atomisticparsers/gromacs/parser.py

```diff
@@ -1690,30 +1722,62 @@ def write_to_archive(self):
                     param.lower() if isinstance(param, str) else param
                 )
 
-        # I have no idea if output trajectory file can be specified in input
-        trr_file = self.get_gromacs_file('trr')
-        trr_file_nopath = trr_file.rsplit('.', 1)[0]
-        trr_file_nopath = trr_file_nopath.rsplit('/')[-1]
-        xtc_file = self.get_gromacs_file('xtc')
-        xtc_file_nopath = xtc_file.rsplit('.', 1)[0]
-        xtc_file_nopath = xtc_file_nopath.rsplit('/')[-1]
-        if not trr_file_nopath.startswith(self._basename):
-            trajectory_file = (
-                xtc_file if xtc_file_nopath.startswith(self._basename) else trr_file
-            )
-        else:
-            trajectory_file = trr_file
-
         self.traj_parser.mainfile = topology_file
-        self.traj_parser.auxilliary_files = [trajectory_file]
-
-        # check to see if the trr file can be read properly (and has positions), otherwise try xtc file instead
-        positions = None
-        if (universe := self.traj_parser.universe) is not None:
-            atoms = getattr(universe, 'atoms', None)
-            positions = getattr(atoms, 'positions', None)
-        if positions is None:
-            self.traj_parser.auxilliary_files = [xtc_file] if xtc_file else [trr_file]
+        # JFR comment: I have no idea if output trajectory file can be specified in input
+        # ? Do you mean, e.g., a flag in the *.log file that specifies the output trajectory format?
+        self.traj_file_priority_list = ['trr', 'xtc', 'pdb', 'gro']
+        self.traj_parser.auxilliary_files = self.find_trajectory_files()
+        if self.traj_parser.auxilliary_files:
+            # Iterate through available trajectory formats until a valid one is found
+            while self.traj_file_priority_list:
```

Comment:
As indicated below, I would make this a for loop over some sort of global priority list of files.